keras_nlp/utils/preset_utils.py

```diff
@@ -361,51 +383,47 @@ def upload_preset(
         )
 
 
-def load_from_preset(
-    preset,
-    load_weights=True,
-    config_file="config.json",
-    config_overrides={},
-):
-    """Load a KerasNLP layer to a preset directory."""
-    # Load a serialized Keras object.
+def load_config(preset, config_file=CONFIG_FILE):
     config_path = get_file(preset, config_file)
     with open(config_path) as config_file:
         config = json.load(config_file)
-    config["config"] = {**config["config"], **config_overrides}
-    layer = keras.saving.deserialize_keras_object(config)
+    return config
 
-    # Load any assets for our tokenizers.
-    tokenizer = get_tokenizer(layer)
-    if tokenizer and config["assets"]:
-        for asset in config["assets"]:
-            get_file(preset, asset)
-        config_dir = os.path.dirname(config_path)
-        asset_dir = os.path.join(config_dir, TOKENIZER_ASSET_DIR)
-        tokenizer.load_assets(asset_dir)
-
-    # Optionally load weights.
-    load_weights = load_weights and config["weights"]
-    if load_weights:
-        # For jax, delete all previous allocated memory to avoid temporarily
-        # duplicating variable allocations. torch and tensorflow have stateful
-        # variable types and do not need this fix.
-        if backend_config.backend() == "jax":
-            for weight in layer.weights:
-                if getattr(weight, "_value", None) is not None:
-                    weight._value.delete()
-        weights_path = get_file(preset, config["weights"])
-        layer.load_weights(weights_path)
-
-    return layer
+
+def load_serialized_object(
+    preset,
+    config_file=CONFIG_FILE,
+    config_overrides={},
+):
+    config = load_config(preset, config_file)
+    config["config"] = {**config["config"], **config_overrides}
+    return keras.saving.deserialize_keras_object(config)
 
 
 def check_config_class(
     preset,
-    config_file="config.json",
+    config_file=CONFIG_FILE,
 ):
     """Validate a preset is being loaded on the correct class."""
     config_path = get_file(preset, config_file)
     with open(config_path) as config_file:
         config = json.load(config_file)
     return keras.saving.get_registered_object(config["registered_name"])
+
+
+def get_asset_dir(
```

Comment:
i'm not sure this util needs to exist. see comment above.

Reply:
Removed it!